infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -75,45 +80,32 @@ void handleQuery(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext
                 .payloadMsg(new OrchestratorResponse(resp)));
     }
 
-    void addNode(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
-        AddNodeRequest req = (AddNodeRequest) msg.getPayload().getRequest();
-        if (findUUID(req.getEndpoint()) != null) {
-            // An add node workflow is already executing for this endpoint, return
+    synchronized void dispatch(CorfuPayloadMsg<OrchestratorRequest> msg,
```

Comment:
Should have a javadoc too. Please also describe why it needs to be synchronized.

Reply:
done.